internal/mapping.go

```diff
@@ -164,3 +167,60 @@ func GetSpannerKeyName(srcKeyName string, schemaForeignKeys map[string]bool) str
 	schemaForeignKeys[spKeyName] = true
 	return spKeyName
 }
+
```

Comment:
Thanks for uncovering the overall issue here with table conversion order --- good find!
And thanks for coming up with this approach -- I agree this is a good way to tackle the issue.